src/getThreadPictures.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function getThreadPictures(threadID, offset, limit, callback) {
+    if(!callback) {
+      callback = function() {};
+    }
+
+    var form = {
+      'thread_id': threadID,
+      'offset': offset,
+      'limit': limit
+    };
+
+    defaultFuncs
+      .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+        if (resData.error) {
+          throw resData;
+        }
+        return Promise.all(resData['payload']['imagesData'].map(function(image) {
+          return new Promise(function(resolve, reject) {
+            form = {
+              'thread_id': threadID,
+              'image_id': image.fbid
+            };
+            defaultFuncs
+            .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+            .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+            .then(function(resData) {
+              if (resData.error) {
+                throw resData;
+              }
+              // the response is pretty messy
+              var queryThreadID = resData.jsmods.require[0][3][1].query_metadata.query_path[0].message_thread;
+              var imageData = resData.jsmods.require[0][3][1].query_results[queryThreadID].message_images.edges[0].node.image2;
```

Comment:
I'm not sure about `edges`, it is just part of the response:
```
"edges": [{
    "node": {
        "id": "10207439106202914",
        "image1": {
            "uri": "https:\/\/scontent.xx.fbcdn.net\/v\/t34.0-12\/13734559_10207439106202914_2090663689_n.png?oh=879662957f0cc6733705ccc45fa8c242&oe=579A5ECD",
            "width": 960,
            "height": 355
        },
        "image2": {
            "uri": "https:\/\/scontent.xx.fbcdn.net\/v\/t35.0-12\/s2048x2048\/13833333_10207439106202914_2090663689_o.png?oh=6b4e847e5de29530f5a7e87d2e52f082&oe=579956A8",
            "width": 2048,
            "height": 757
        }
    },
    "cursor": "10207439106202914"
}],
```
As for `image2`, there always seems to be `image1` and `image2` returned in `node`, but `image2` is the one with the greater dimensions and correct original size, which is why I chose to return that value.